app/modules/search/controllers/searchControllers.js

```diff
@@ -21,6 +21,183 @@
+                            break;
+                    }
+                }
+            },
+            keyword: {
+                label: 'Keyword',
+                field: ["keywords"],
+                size: 200,
+                order: 2
+            }
+        };
+
+        normalizeAggregationFilter = function(){
+            var filters = [];
+            angular.forEach($scope.facetCategories, function (aggregation){
+                var terms = [];
+                angular.forEach(aggregation.buckets, function(bucket){
+                    if (bucket.selected){
+                        var field = aggregations[aggregation.name].field;
+                        if (angular.isArray(field)) {
+                            angular.forEach(field, function (fld) {
+                                var term = {};
+                                term[fld] = bucket.value;
+                                terms.push({"term":term});
+                            });
+                        } else {
+                            var term = {};
+                            term[field] = bucket.value;
+                            terms.push({"term":term});
+                        }
+                    }
+                });
+                if (terms.length>0){
+                    filters.push({"bool":{"should":terms}});
+                }
+            });
+            if (filters.length>0)
+                return {"bool":{"must":filters}};
+            return {};
+        };
+        $scope.facetCategoryResults = function(name){
+            return 1;
+            if ($scope.aggregationData[name])
+                return $scope.aggregationData[name].length;
+            return 0;
+        };
+        prepareAggregations = function(){
+            var request = {};
+            angular.forEach(aggregations, function (aggregation, name){
+                $scope.facetCategories.push({
+                    name: name,
+                    label: aggregation.label,
+                    order: aggregation.order,
+                    results: 0,
+                    buckets: {}
+                });
+                aggregations[name].index = $scope.facetCategories.length-1;
+
+                if (!angular.isArray(aggregation.field))
+                    aggregations[name].field = [aggregation.field];
+
+                var i=0;
+                angular.forEach(aggregation.field, function (fld){
+                    var obj = {};
+                    obj.field = fld;
+                    if (aggregation.size)
+                        obj.size = aggregation.size;
+                    var index = (aggregation.field.length>1)?name+'$'+(i++):name;
+                    request[index] = {
+                        "terms" : obj
+                    }
+                });
+            });
+            requestAggs = request;
+        };
+
+        normalizeAggregationResults = function(aggs){
+            //console.log(aggs);
```

Comment:
yes there are others too, like in line 187, but it's there if needed for debugging. I'm not sure what's the issue with those? would you prefer to leave the log messages on or not include these at all?